project/plugins.sbt

```diff
@@ -3,20 +3,26 @@ resolvers += Resolver.url("HMRC-open-artefacts-ivy2", url("https://open.artefact
   Resolver.ivyStylePatterns
 )
 
-(managedSources in Compile) += (baseDirectory.value / "project" / "PlayCrossCompilation.scala")
+sys.env.get("PLAY_VERSION") match {
+  case Some("2.8") => // required since we're cross building for Play 2.8 which isn't compatible with sbt 1.9
+    libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
+  case _           => libraryDependencySchemes := libraryDependencySchemes.value // or any empty DslEntry
+}
+
+sys.env.get("PLAY_VERSION") match {
+  case Some("2.8") => addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.20")
+  case Some("2.9") => addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.9.0")
+  case _           => addSbtPlugin("org.playframework" % "sbt-plugin" % "3.0.0")
+}
```

Comment:
not worth doing anything more now - but one thing that I'm still not sure of with all this is whether play-nunjucks itself needs to have a dependency on play itself - or if this is just here because of the integration projects

for example, we still have to run the build multiple times with different PLAY_VERSIONs in play-frontend-hmrc because we have routes that needs to be compiled and generate reverse routes - however bootstrap refactored their compiled routes to just be a router (because they didn't need reverse routing) and so they can just run the build once and all the different versions of the library get built at the same time for different versions of play because they don't depend on play plugin. I wonder if play-nunjucks could go further to remove the PLAY_VERSION stuff because it seems like it might just be using the play plugin for the integration sub projects - but that's total "future" stuff

Reply:
@oscarduignan That's some really good thoughts. I think it's worth capturing these on our upgrading notes document.